File: app/src/main/java/com/pietrantuono/podcasts/singlepodcast/customviews/EpisodesAdapter.java
package com.pietrantuono.podcasts.singlepodcast.customviews;

import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.pietrantuono.podcasts.R;
import com.pietrantuono.podcasts.apis.PodcastEpisode;
import com.pietrantuono.podcasts.imageloader.SimpleImageLoader;

import java.util.ArrayList;
import java.util.List;

public class EpisodesAdapter extends RecyclerView.Adapter<EpisodeHolder> {
    private final List<PodcastEpisode> items;
    private final SimpleImageLoader simpleImageLoader;

    public EpisodesAdapter(SimpleImageLoader simpleImageLoader) {
        this.simpleImageLoader = simpleImageLoader;
        items = new ArrayList<>();
    }

    @Override
    public EpisodeHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext())
                .inflate(R.layout.episode_item, parent, false);
        EpisodeHolder holder = new EpisodeHolder(view);
        return holder;
    }

    @Override
    public void onBindViewHolder(EpisodeHolder holder, int position) {
        holder.bind(items.get(position));
    }

    @Override
    public int getItemCount() {
        return items.size();
    }

    public void setItems(List<PodcastEpisode> items) {
        this.items.clear();
        this.items.addAll(items);
        notifyDataSetChanged();
    }
}

File: app/src/main/java/com/pietrantuono/podcasts/singlepodcast/customviews/EpisodesAdapter.kt
package com.pietrantuono.podcasts.addpodcast.singlepodcast.customviews

import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.ViewGroup
import com.pietrantuono.podcasts.R
import com.pietrantuono.podcasts.addpodcast.singlepodcast.viewmodel.ResourcesProvider
import models.pojos.Episode
import java.util.*

class EpisodesAdapter(private val resourcesProvider: ResourcesProvider) : RecyclerView.Adapter<EpisodeHolder>() {
    private val items: MutableList<Episode>
    private var clickListener: OnItemClickListener? = null

    init {
        items = ArrayList()
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): EpisodeHolder {
        val view = LayoutInflater.from(parent.context)
                .inflate(R.layout.episode_item, parent, false)
        return EpisodeHolder(view, resourcesProvider)
    }

    override fun onBindViewHolder(holder: EpisodeHolder, position: Int) {
        holder.bind(items[position], clickListener)
    }

    override fun getItemCount(): Int {
        return items.size
    }

    fun setItems(items: List<Episode>) {
        this.items.clear()
        this.items.addAll(items)
        notifyDataSetChanged()
    }

    fun setOnItemClickListener(clickListener: OnItemClickListener) {
        this.clickListener = clickListener
    }


    interface OnItemClickListener {
        fun onItemClick(episode: Episode)
    }
}
